src/components/Input/Input.tsx

```diff
@@ -0,0 +1,180 @@
+
+  ${(p) =>
+    p.hasError &&
+    css`
+      color: ${Color.ErrorRed};
+    `}
+  /* TODO remove 'all', animate only 'fontsize' and 'top'*/
+    transition: font-size 0.3s ease;
+`;
+
+const LabelContainer = styled.div<LabelContainerProps>`
+  display: flex;
+  flex-direction: row;
+  pointer-events: none;
+  position: absolute;
+  align-items: center;
+  z-index: 6;
+  pointer-events: ${(p) => (p.disabled ? "none" : "auto")};
+  padding-left: 16px;
+
+  ${(props) =>
+    props.hasFocus &&
+    css`
+      top: ${(p: { size: number }) => `-${p.size / 4 - 3}px`};
+      padding-left: 5px;
+      margin-left: 11px;
+      background: ${Color.BackgroundMain};
+      height: 16px;
+
+      ${StyledLabel} {
+        font-size: 14px;
+        padding-left: 0;
+        padding-top: 0;
+        padding-bottom: 0;
+      }
+
+      svg {
+        background: ${Color.BackgroundMain};
+        transform: scale(0.85);
+        transition: transform 0.3s ease;
+      }
+    `};
+`;
+
+const ErrorText = styled.h1`
+  position: absolute;
+  bottom: 0;
+  font-family: sans-serif;
+  font-size: 14px;
+  font-weight: normal;
+  color: ${Color.ErrorRed};
+`;
+
+const StyledInput = styled.input<StyledInputProps>`
+  position: relative;
+  height: ${(p) => `${p.size}px`};
+  width: 100%;
+  padding: 16px;
+  border: 2px solid ${Color.BorderGrey};
+  border-radius: 7px;
+  background: ${Color.BackgroundMain};
+  font-size: 17px;
+  cursor: ${(p) => (p.disabled ? "default" : "pointer")};
+
+  :disabled {
+    border: 2px solid ${Color.BorderDisabled};
+    background: ${Color.BackgroundDisabled};
+  }
+`;
+
+const IconContainer = styled(Row)<{ disabled?: boolean }>`
+  ${(p) =>
+    p.disabled &&
+    css`
+      svg * {
+        fill: ${Color.DisabledGrey};
+      }
+    `}
+`;
+
+// Pass accurate input type
+// https://www.w3.org/WAI/tutorials/forms/validation/#validating-common-input
+
+export const Input: FC<InputProps> = ({
+  className,
+  disabled = false,
+  label,
+  onInputChange,
+  name,
+  tabIndex,
+  type = "text",
+  required,
+  maxLength,
+  pattern,
+  size = "large",
+  width = 300,
+  icon,
+  hasError = false,
+  isValid = false,
+  errorText = "Input not valid",
+  value,
+}: InputProps) => {
+  const [hasFocus, setHasFocus] = useState<boolean>(false);
+
+  // TODO - don't use disabled, similar to Button.tsx
+  // 'disabled' removes element for screen readers, so for a11y it's best to visually make them disbaled and use aria-diabled instead - // aria-disabled={disabled}
+  // https://a11y-101.com/development/aria-disabled
+
+  const sizeNumber = size === "medium" ? 50 : 56;
+
+  const setFocus = () => {
+    setHasFocus(true);
+  };
+
+  const removeFocus = () => {
+    setHasFocus(false);
+  };
+
+  return (
+    <Column className={className} minHeight={sizeNumber + ERROR_TEXT_HEIGHT} position="relative" width={width}>
+      <LabelContainer disabled={disabled} hasFocus={hasFocus || value} hasIcon={icon !== "undefined"} size={sizeNumber}>
```

Comment:
If we are not using `hasIcon` prop in the `<LabelContainer/>` Styled Component please remove it and the type-checking.